audit.sh

```diff
@@ -0,0 +1,406 @@
+
+    counter=1
+
+    if [ $( echo $_files | wc -w) -le 1 ]; then
+        selection="$_files"
+    else
+        echo
+        for i in $_files; do
+            echo $counter" -" $i
+            counter=$((counter + 1))
+        done
+        echo "Which option number would you like to check? "
+        read input </dev/tty
+        echo
+        selection=$(echo $_files | cut -d " " -f $input)
+        echo "Your option was "$input "which is ""$selection"
+    fi
+}
+
+# Checks repo master branch with cloned repo commit hash - verify sha1sum
+validate_current_commit() {
+    _trailingslash="$_pwd/$1/.gitrepo"
+    _notrailingslash="$_pwd/$1.gitrepo"
+    repo_location="$2"
+
+    if [ -a "$_notrailingslash" 2>/dev/null ]; then
+        _subrepo_information="$_notrailingslash"
+        _subrepo_name="$1"
+        _dir="$_pwd/$1"
+    elif [ -a "$_trailingslash" 2>/dev/null ]; then
+        _subrepo_information="$_trailingslash"
+        _subrepo_name="$1/"
+        _dir="$_pwd/$1/"
+    else
+        echo
+        echo -e "$RED""$1.gitrepo" "repo does not appear to exist.""$EFC"
+        echo "You can only check for updates against the master branch"
+        exit 1
+    fi
+
+    find_checksum_file "$_pwd/$1"
+
+    case "$repo_location" in
+        "verify")  # Verifying the commit hash is valid
+        branch_name=$(git branch | grep \* | cut -d ' ' -f2)
+        repo_location="remote"
+
+        raw_subrepo="https://raw.githubusercontent.com/rackerlabs/useful-scripts/$branch_name/$_subrepo_name"
+        subrepo_sha1sum=$(curl -s "$raw_subrepo$selection" | sha1sum | awk '{print $1}' )
+        script_location="$raw_subrepo$selection"
+
+        commit_path="$raw_subrepo"".gitrepo"
+        _commit=$(curl -s "$commit_path" | awk -F' = ' '/commit/ { print $2 }')
+        ;;
+
+        "local")  # For branches not yet pushed to github
+        branch_name=$(git branch | grep \* | cut -d ' ' -f2)
+
+        subrepo_sha1sum=$(sha1sum "$_dir$selection" | awk '{print $1}' )
+        script_location="$_dir$selection"
+
+        commit_path="$_dir"".gitrepo"
+        _commit=$(awk -F'= ' '/commit/ {print $2}' "$_subrepo_information")
+        _remote=$(awk -F'= ' '/remote/ {print $2}' "$_subrepo_information")
+        ;;
+
+        *)
+        branch_name="master"
+        repo_location="remote"
+
+        raw_subrepo="https://raw.githubusercontent.com/rackerlabs/useful-scripts/$branch_name/$_subrepo_name"
+        subrepo_sha1sum=$(curl -s "$raw_subrepo$selection" | sha1sum | awk '{print $1}' )
+
+        script_location="$raw_subrepo$selection"
+        commit_path="$raw_subrepo"".gitrepo"
+        _commit=$(curl -s "$commit_path" | awk -F' = ' '/commit/ { print $2 }')
+        ;;
+    esac
+
+    _remote=$(awk -F'= ' '/remote/ {print $2}' "$_subrepo_information")
+
+    # Find the sha1sum of the original script
+    raw_remote=$(echo $_remote | sed 's/github.com/raw.githubusercontent.com/g'  | sed "s/\.git$/\/$_commit/g")
```

Comment:
We may be able to use something like `${variable//search/replace}` here instead of sed - not tested this though. Also could probably be applied elsewhere in this.